### PR TITLE
TYP: Add type hints and improve docstrings of load_tile_map function

### PR DESCRIPTION
**Pull request by seisman**
**Preview**: https://pygmt-dev--3087.org.readthedocs.build/en/3087/api/generated/pygmt.datasets.load_tile_map.html#pygmt.datasets.load_tile_map

### CODE REVIEW

**Review by michaelgrund — APPROVED** [review]

File: pygmt/datasets/tile_map.py [hunk 1]
```diff
@@ -100,15 +89,15 @@ def load_tile_map(
...
     Returns
     -------
-    raster : xarray.DataArray
+    raster
         Georeferenced 3-D data array of RGB values.
 
     Raises
     ------
     ImportError
         If ``contextily`` is not installed or can't be imported. Follow
```
Comment (yvonnefroehlich) [inline pygmt/datasets/tile_map.py:98]:
```suggestion
        If ``contextily`` is not installed or can't be imported. Follow the
```

File: pygmt/datasets/tile_map.py [hunk 2]
```diff
@@ -100,15 +89,15 @@ def load_tile_map(
 
     Returns
     -------
-    raster : xarray.DataArray
+    raster
         Georeferenced 3-D data array of RGB values.
```
[the remark this answers is not in the document]
Comment (seisman) [reply pygmt/datasets/tile_map.py:93]: Need to check if sphinx-autodoc-typehint provides an option to control the behavior. https://github.com/tox-dev/sphinx-autodoc-typehints#options

[the remark this answers is not in the document]
Comment (seisman) [reply pygmt/datasets/tile_map.py:93]: I don't think sphinx-autodoc-typehint provides such an option, so there is little we can do unless we decide not to add type hints for the return values.

[the remark this answers is not in the document]
Comment (weiji14) [reply pygmt/datasets/tile_map.py:93]: There was a `napoleon_use_rtype = True` option that was tried in https://github.com/GenericMappingTools/pygmt/issues/937. Does it work here?

[the remark this answers is not in the document]
Comment (weiji14) [reply pygmt/datasets/tile_map.py:93]: Can you try `napoleon_preprocess_types = True` as suggested at https://github.com/sphinx-doc/sphinx/issues/9394?

[the remark this answers is not in the document]
Comment (seisman) [reply pygmt/datasets/tile_map.py:93]: `napoleon_use_rtype = True` incorrectly parses `raster` as the return type:
Screenshot from 2024-03-08 08-27-45

`napoleon_preprocess_types = True` and `napoleon_use_rtype = False` is not bad:

Screenshot from 2024-03-08 08-29-06

[the remark this answers is not in the document]
Comment (seisman) [reply pygmt/datasets/tile_map.py:93]: https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html#confval-napoleon_preprocess_types

> napoleon_preprocess_types
>    True to convert the type definitions in the docstrings as references. Defaults to False.

When setting `napoleon_preprocess_types = True`, `raster` is parsed as type definitions, so it's rendered the same way as the return type `DataArray`. Strictly speaking, it's still incorrect.

Actually, the name of the returned value should not appear in the NumPy-style docstrings (see https://www.sphinx-doc.org/en/master/usage/extensions/example_numpy.html#example-numpy). I.e., we should change
```
    Returns
    -------
    raster
        Georeferenced 3-D data array of RGB values.
```
to 
```
    Returns
    -------
    Georeferenced 3-D data array of RGB values.
```

[the remark this answers is not in the document]
Comment (weiji14) [reply pygmt/datasets/tile_map.py:93]: > Actually, the name of the returned value should not appear in the NumPy-style docstrings (see https://www.sphinx-doc.org/en/master/usage/extensions/example_numpy.html#example-numpy). I.e., we should change

It looks like they are using something like

```
    Returns
    -------
    xr.DataArray
        Georeferenced 3-D data array of RGB values.
```

[the remark this answers is not in the document]
Comment (seisman) [reply pygmt/datasets/tile_map.py:93]: That version doesn't have type hints for the return value.

[the remark this answers is not in the document]
Comment (seisman) [reply pygmt/datasets/tile_map.py:93]: Some related discussions:

- https://github.com/sphinx-doc/sphinx/issues/7077
- https://github.com/tox-dev/sphinx-autodoc-typehints/pull/311
- https://github.com/numpy/numpydoc/issues/356

**Comment by seisman** [comment]
I'm merging this PR. The style issue of the function return value/type should be addressed separately (if there is a good solution).

**Pull request merged**